sormas-api/src/main/java/de/symeda/sormas/api/report/AggregatedCaseCountDto.java

```diff
@@ -291,39 +237,181 @@ public void setEpiWeek(Integer epiWeek) {
+			&& Objects.equals(pointOfEntryName, that.pointOfEntryName)
+			&& Objects.equals(pointOfEntryId, that.pointOfEntryId)
+			&& disease == that.disease
+			&& Objects.equals(ageGroup, that.ageGroup);
+	}
+
+	@Override
+	public int hashCode() {
+		return Objects.hash(
+			regionName,
+			regionId,
+			districtName,
+			districtId,
+			healthFacilityName,
+			healthFacilityId,
+			pointOfEntryName,
+			pointOfEntryId,
+			year,
+			epiWeek,
+			disease,
+			newCases,
+			labConfirmations,
+			deaths,
+			ageGroup);
+	}
+
+	public boolean similar(Object o, AggregateReportGroupingLevel groupingLevel) {
+
+		if (this == o)
+			return true;
+		if (o == null || getClass() != o.getClass())
 			return false;
-		if (labConfirmations != other.labConfirmations)
+		AggregatedCaseCountDto that = (AggregatedCaseCountDto) o;
+		boolean similar = year == that.year && epiWeek == that.epiWeek && disease == that.disease && Objects.equals(ageGroup, that.ageGroup);
+		if (groupingLevel != null) {
+			switch (groupingLevel) {
+
+			case REGION:
+				similar = similar && Objects.equals(regionName, that.regionName) && Objects.equals(regionId, that.regionId);
+				break;
+			case DISTRICT:
+				similar = similar && Objects.equals(districtName, that.districtName) && Objects.equals(districtId, that.districtId);
+				break;
+			case HEALTH_FACILITY:
+				similar =
+					similar && Objects.equals(healthFacilityName, that.healthFacilityName) && Objects.equals(healthFacilityId, that.healthFacilityId);
+				break;
+			case POINT_OF_ENTRY:
+				similar = similar && Objects.equals(pointOfEntryName, that.pointOfEntryName) && Objects.equals(pointOfEntryId, that.pointOfEntryId);
+				break;
+			default:
+			}
+		}
+		return similar;
+	}
+
+	public boolean equalJurisdiction(Object o) {
+		if (this == o)
+			return true;
+		if (o == null || getClass() != o.getClass())
 			return false;
-		if (newCases != other.newCases)
+		AggregatedCaseCountDto that = (AggregatedCaseCountDto) o;
+		return Objects.equals(regionName, that.regionName)
+			&& Objects.equals(regionId, that.regionId)
+			&& Objects.equals(districtName, that.districtName)
+			&& Objects.equals(districtId, that.districtId)
+			&& Objects.equals(healthFacilityName, that.healthFacilityName)
+			&& Objects.equals(healthFacilityId, that.healthFacilityId)
+			&& Objects.equals(pointOfEntryName, that.pointOfEntryName)
+			&& Objects.equals(pointOfEntryId, that.pointOfEntryId);
+	}
+
+	public boolean higherJurisdictionLevel(Object o) {
+
+		if (this == o)
+			return true;
+		if (o == null || getClass() != o.getClass())
 			return false;
-		if (!Objects.equals(ageGroup, other.ageGroup))
+		AggregatedCaseCountDto that = (AggregatedCaseCountDto) o;
+
+		if (regionId == null && that.regionId != null) {
+			return true;
+		}
+		if (districtId == null && that.districtId != null) {
+			return true;
+		}
+		if (healthFacilityId == null && that.healthFacilityId != null && pointOfEntryId == null && that.pointOfEntryId == null) {
```

Comment:
Probably easier to understand when combining both if statements to this one:

```suggestion
		if (healthFacilityId == null && pointOfEntryId == null && (that.healthFacilityId != null || that.pointOfEntryId != null)) {
```